hawk/app/models/cib.rb

```diff
@@ -1020,7 +1020,12 @@ def initialize(id, user, use_file = false, stonithwarning = false)
       if rsc
         rsc_state = rsc[:state]
       elsif n[:host]
-        rsc_state = CibTools.rsc_state_from_lrm_rsc_op(@xml, n[:host], n[:id])
+	begin
+          rsc_state = CibTools.rsc_state_from_lrm_rsc_op(@xml, n[:host], n[:id])
+        rescue NoMethodError => e
+          # bsc#1163381. Catch the case resource name = node name. this is not allowed in pacemaker
+          error _("please check if any resources have IDs that are conflicting with node names" + "#{e.backtrace.first}: #{e.message} (#{e.class})")
```

Comment:
s/please/Please/ :-)

Not sure about how it's possible in the context here,  but and as mentioned in the bugzilla, could we make an exception so that it doesn't raise the error for `ocf:pacemaker:remote` type of resource? Since in that case, the resource is expected to have the ID that is the exactly consistent with the name of the remote node it handles.

Or if it's a "remote" type of node, we don't raise the error?

Reply:
thx yan for review will have look

Reply:
@gao-yan  this is already present in code:

check
https://github.com/ClusterLabs/hawk/pull/211/files/565605e708cdaadd512f840f2098a1e9162e8c69#diff-1f508dc254aa7d0561bbef5740e5d3deL1018

Reply:
that check is already present at:
https://github.com/ClusterLabs/hawk/pull/211/files/565605e708cdaadd512f840f2098a1e9162e8c69#diff-1f508dc254aa7d0561bbef5740e5d3deL1018

Reply:
Ah, I see. Alright then. Thanks!